src/Microsoft.IdentityModel.Xml/LogMessages.cs

```diff
@@ -73,6 +73,7 @@ internal static class LogMessages
         internal const string IDX30211 = "IDX30211: The TransfromFactory does not support the canonicalizing transform: '{0}'.";
         internal const string IDX30212 = "IDX30212: Unable to verify Signature as Signature.SignedInfo is null.";
         internal const string IDX30213 = "IDX30213: The CryptoProviderFactory: '{0}', CreateForSigning returned null for key: '{1}', SignatureMethod: '{2}'.";
+        internal const string IDX30214 = "IDX30214: User supplied validation of the algorithm '{0}' failed.";
```

Comment:
Could this be reworded as "Algorithm '{0}' did not match any of the valid algorithms supplied by the user. Valid algorithms: {1}.'

Reply:
This is a good idea. It would make debugging easier.

Reply:
Okay, can't quite use that. This capture the failure from Validators.ValidateAcceptedAlgorithms. Which also include the case for the validation failing from the delegate.